File: src/main/java/com/natia/secretmod/utils/Location.java
package com.natia.secretmod.utils;

import com.natia.secretmod.SecretUtils;

import java.util.List;

public enum Location {
    VILLAGE("Village"),
    PRIVATE_ISLAND("Your Island"),
    THE_CATACOMBS("The Catacombs"),
    VOID_SEPULTURE("Void Sepulture"),
    THE_END("The End"),
    NONE("None");
    private String locationName;
    private Location(String locationName) {
        this.locationName = locationName;
    }

    public String asName() {
        return locationName;
    }

    public static Location getAsLocation(String locationName) {
        for (Location location : Location.values()) {
            if (location.asName().contains(locationName)) {
                return location;
            }
        }

        return NONE;
    }

    public static Location getCurrentLocation() {
        try {
            if (SecretUtils.isValid()) {
                List<String> scoreboard = SecretUtils.getScoreboardLines();

                for (String s : scoreboard) {
                    String sCleaned = SecretUtils.cleanSB(s);
                    for (Location location : Location.values()) {
                        if (sCleaned.contains(location.toString()))
                            return location;
                    }
                }
            }

        } catch (Exception e) { e.printStackTrace(); }
        return Location.NONE;
    }

    @Override
    public String toString() {
        return locationName;
    }
}

File: src/main/kotlin/natia/skytweaks/utils/Location.kt
package natia.skytweaks.utils

import natia.skytweaks.SecretUtils

enum class Location private constructor(private val locationName: String) {
    VILLAGE("Village"),
    PRIVATE_ISLAND("Your Island"),
    THE_CATACOMBS("The Catacombs"),
    VOID_SEPULTURE("Void Sepulture"),
    THE_END("The End"),
    NONE("None");

    fun asName(): String {
        return locationName
    }

    override fun toString(): String {
        return locationName
    }

    companion object {

        fun getAsLocation(locationName: String): Location {
            for (location in values()) {
                if (location.asName().contains(locationName)) {
                    return location
                }
            }

            return NONE
        }

        val currentLocation: Location
            get() {
                try {
                    if (SecretUtils.isValid) {
                        val scoreboard = SecretUtils.scoreboardLines

                        for (s in scoreboard) {
                            val sCleaned = SecretUtils.cleanSB(s)
                            for (location in values()) {
                                if (sCleaned.contains(location.toString()))
                                    return location
                            }
                        }
                    }

                } catch (e: Exception) {

                }

                return Location.NONE
            }
    }
}
